book/version_control.md

```diff
@@ -58,91 +52,110 @@ You should **not** include the following in your code repository:
 
-### Git Commands
+```{figure} ./_static/git_feature.png
+---
+width: 70%
+name: git_feature
+alt: Creating a feature branch from a main branch.
+---
+Working on changes on a single `feature` branch.
+```
 
-Git commands are run from the terminal, which is typically [bash](https://en.wikipedia.org/wiki/Bash_(Unix_shell)) or a command prompt.
+Here we show a single `feature` branch being created from the `main` branch. The work is initially quite experimental, but is refined over a few commits. The complete, working feature is finally merged back onto the `main` branch.
 
-Commands have a very standard structure in Git, namely they always starts with `git` to run the command using the Git program.
+Many small scale projects iteratively work on individual feature or development branches in this way. The [GitHub flow branching strategy](https://guides.github.com/introduction/flow/) uses this approach in combination with [Pull Requests](pull-requests), to incorporate peer review into the development workflow.
 
-Git commands are also made up of:
-* a command name, which always follows `git` (e.g. `git status`)
-* command arguments, which follow the command and can be put before or after any flags (e.g. `git clone https://github.com/pandas-dev/pandas.git`)
-* optional flags
-  * these can be long, starting with two dashes (e.g. `--no-edit`)
-  * or short, starting with one dash (e.g. `-m`)
-  *  note that the `-h` flag can be used to display help for any command
-* flag arguments, which follow some flags that take arguments (e.g. `git commit -m "my message"`)
 
-You can use this [cheat-sheet](https://education.github.com/git-cheat-sheet-education.pdf) as a reference for the most common Git commands.
+```{figure} ./_static/git_develop.png
+---
+width: 90%
+name: git_develop
+alt: Iteratively creating and merging a develop branch from a main branch.
+---
+Iteratively working on a `develop` branch.
+```
 
+We can create multiple branches from our `main` branch at any point. We might do this when multiple features are being developed in parallel, or perhaps when multiple analysts wish to make changes to the same piece of code independently.
 
-### Git versioning concepts
+```{figure} ./_static/git_multiple_features.png
+---
+width: 90%
+name: git_multiple_features
+alt: Two features branches created from a single main branch.
+---
+Working on multiple parallel branches.
+```
+
+Here we create two feature branches from `main`. Work on each feature is carried out independently of the other feature and can be merged onto `main` once it is complete. If changes from separate branches affect the same files, merging these branches to `main` may lead to merge conflicts. In these cases you should ensure that you resolve the conflicts to keep the desired overall change.
```

Comment:
Great place to maybe say: 
```
if you are able to break up your work into independent features that are not expected to affect the same files, do so as this helps avoid conflicts in the merging process that need to be resolved manually 
```

also for context its worth showing or explaining how to deal with these conflicts - best practice in resolving merge conflicts.

Reply:
Also how do I know that after a merge has happened my code still works when a person removed and assumed things away ?